### PR TITLE
EDU-549 Introduce storage context 

### PR DESCRIPTION
**Pull request by adrianaluchian**
Closes https://educandu.atlassian.net/browse/EDU-549

### CODE REVIEW

File: src/utils/urls.js [hunk 1]
```diff
@@ -157,8 +157,21 @@ function getLessonUrl({ id, slug, view, templateLessonId }) {
   return queryString ? `${url}?${queryString}` : url;
 }
 
+function getDocIdIfDocUrl(url) {
+  const docPageRegex = new RegExp(`^(?:${docsPrefix})([a-zA-Z0-9]+)\\b`, 'i');
```
Comment (ahelmberger) [inline src/utils/urls.js:161]: We could extract those two regexes into a module level variable to have them closer to where also `docsPrefix` and `lessonsPrefix` live and be created just once (as they're called for each request). Also, maybe, although we know that `docsPrefix` and `lessonsPrefix` at the moment do not contain characters that would have to be escaped for use in a regex, it wouldn't hurt escaping them anyways.

**Comment by ahelmberger** [comment]
Really nice that this takes a lot of pollution out of so many places in the code (especially plugins) :heart_eyes:

File: src/components/storage-context.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+StorageProvider.propTypes = {
+  children: PropTypes.node,
+  value: storageShape.isRequired
+};
+
+StorageProvider.defaultProps = {
+  children: null
+};
+
+export default {
```
[the remark this answers is not in the document]
Comment (adrianaluchian) [reply src/components/storage-context.js]: Indeed, and as it happens all contexts followed this same pattern and from none of them was the default export ever used. Cleaned them all, thanks for the hint 👍

**Comment by adrianaluchian** [comment]
> Really nice that this takes a lot of pollution out of so many places in the code (especially plugins) 😍

Oh yes, and the locations setup logic is all in one place finally, nice, controllable and testable 😁

File: src/utils/urls.js (hunk 1 above)
Comment (adrianaluchian) [reply src/utils/urls.js]: Good point, extracted them both to module level and escaped prefixes.

File: src/components/storage-context.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (ahelmberger) [reply src/components/storage-context.js]: Cool :+1:

**Review by ahelmberger — APPROVED** [review]

**Pull request merged**